### PR TITLE
26 05 2021

### PR DESCRIPTION
**Pull request by RS-lalit-tyagi**

### CODE REVIEW

File: dbHelper/admin.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+				name, 
+				phone, 
+				email, 
+				position,
+				profile_image,
+				created_at
+			FROM users 
+			WHERE archived_at IS NULL OFFSET $1 LIMIT $2`
+	var users []models.User
+	rows, err := database.RemoteOfficeDB.Query(SQL, offset, limit)
```
Comment (tejashwikalptaru) [inline dbHelper/admin.go:24]: add join with images, do not run query multiple times for images in `rows`

File: dbHelper/admin.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+func UpdateAllowedLeaves(al models.AllowedLeave) error {
+	SQL := `UPDATE allowed_leaves SET type=$1 ,year=$2,allowed=$3 	WHERE  id = $4`
+	_, err := database.RemoteOfficeDB.Exec(SQL, al.LeaveType, al.Year, al.Allowed, al.ID)
+	return err
+}
+
+//GetAllowedLeaveByID return allowed leave model for given leav id
+func GetAllowedLeaveByID(leaveID int) (models.AllowedLeave, error) {
+	SQL := `SELECT * FROM allowed_leaves where id = $1`
+	leaveAllowedDetails := models.AllowedLeave{}
```
Comment (tejashwikalptaru) [inline dbHelper/admin.go:161]: same here

File: handler/admin.go [hunk 3]
```diff
@@ -1,10 +1,194 @@
...
+	"strings"
+	"time"
 )
+
 // Check to test the admin route
 func Check(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
-	_, _ = w.Write([]byte("ok"))}
+	_, _ = w.Write([]byte("ok"))
```
Comment (tejashwikalptaru) [inline handler/admin.go:19]: remove this check handler

File: dbHelper/admin.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+		return nil, err
+	}
+	defer func() {
+		if err := rows.Close(); err != nil {
+			logrus.Error(err)
+		}
+	}()
+	users = make([]models.User, 0)
+	for rows.Next() {
+		user := models.User{}
```
[the remark this answers is not in the document]
Comment (RS-lalit-tyagi) [reply dbHelper/admin.go:35]: done

[the remark this answers is not in the document]
Comment (RS-lalit-tyagi) [reply dbHelper/admin.go:35]: done

File: dbHelper/admin.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return standUps, nil
+}
+
+// GetAllowedLeaves return array of allowed leaves for the given year for all leave types
+func GetAllowedLeaves(year int) ([]models.AllowedLeave, error) {
+	SQL := `SELECT * FROM allowed_leaves where year = $1 ORDER BY type`
+	leaveAllowedDetails := []models.AllowedLeave{}
```
[the remark this answers is not in the document]
Comment (RS-lalit-tyagi) [reply dbHelper/admin.go:139]: FIXED

File: handler/admin.go (hunk 3 above)
Comment (RS-lalit-tyagi) [reply handler/admin.go:19]: done

File: dbHelper/admin.go (hunk 2 above)
Comment (RS-lalit-tyagi) [reply dbHelper/admin.go:161]: done

File: utils/utils.go [hunk 6]
```diff
@@ -174,7 +175,7 @@ func UploadFile(file multipart.File, filename, bucket string) (string, error) {
 		}
 	}()
 
-	ctx, cancel := context.WithTimeout(ctx, time.Second*50)
+	ctx, cancel := context.WithTimeout(ctx, time.Second*180)
```
[the remark this answers is not in the document]
Comment (RS-lalit-tyagi) [reply utils/utils.go:178]: need help

**Pull request closed without merging**